bids/layout/utils.py

```diff
@@ -0,0 +1,86 @@
+import json
+from pathlib import Path
+from packaging.version import Version
+
+
+# As per https://bids.neuroimaging.io/bids_spec1.1.1.pdf
+desc_fields = {
+    Version("1.1.1"): {
+        "required": ["Name", "BIDSVersion"],
+        "recommended": ["License"],
+        "optional": ["Authors", "Acknowledgements", "HowToAcknowledge",
+                     "Funding", "ReferencesAndLinks", "DatasetDOI"]
+    }
+}
+
+
+def get_description_fields(version, type_):
+    if isinstance(version, str):
+        version = Version(version)
+    if not isinstance(version, Version):
+        raise TypeError("Version must be a string or a packaging.version.Version object.")
+
+    if version in desc_fields:
+        return desc_fields[version][type_]
+    return desc_fields[max(desc_fields.keys())][type_]
+
+
+def write_derivative_description(source_dir, name, bids_version='1.1.1', exist_ok=False,
+                                 propagate=False, **desc_kwargs):
+    """Write a dataset_description.json file for a new derivative folder.
+
+    Parameters
+    ----------
+    source_dir : str or Path
+        Directory of the BIDS dataset that has been derived.
```

Comment:
I don't think the original source_dir should be required, or at least it should not be assumed that the derivative dataset will be nested within it. 

Derivative datasets can be live outside of the source_dir in the filesystem.

I also found the description confusing:

```suggestion
        Directory of the source BIDS dataset from which new dataset is derived.
```